File: stian/day_two/go/day_two.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("day_two.txt")
	if err != nil {
		panic(err)
	}

	defer file.Close()
	scanner := bufio.NewScanner(file)
	var data []string

	for scanner.Scan() {
		data = append(data, scanner.Text())
	}
	// var line int
	// data := make([]int, 0)
	// _, err = fmt.Fscanf(file, "%d", &line)

	// for err == nil {
	// 	data = append(data, line)
	// 	_, err = fmt.Fscanf(file, "%d\n", &line)
	// }

	fmt.Println("Part 1:", partOne(data))
	fmt.Println("Part 2:", partTwo(data))
}

func partOne(data []string) int {
	horizontal := 0
	depth := 0
	for _, v := range data {
		fields := strings.Fields(v)
		dir := fields[0]
		num, _ := strconv.Atoi(fields[1])
		if dir == "forward" {
			horizontal += num
		} else if dir == "down" {
			depth += num
		} else {
			depth -= num
		}
	}
	// 2147104
	return horizontal * depth
}

func partTwo(data []string) int {
	horizontal := 0
	depth := 0
	aim := 0
	for _, v := range data {
		fields := strings.Fields(v)
		dir := fields[0]
		num, _ := strconv.Atoi(fields[1])
		if dir == "forward" {
			horizontal += num
			depth += (aim * num)
		} else if dir == "down" {
			aim += num
		} else {
			aim -= num
		}
	}
	// 2044620088
	return horizontal * depth
}

File: stian/day_two/python/day_two.py
data = []
test_data = []
with open('day_two.txt', 'r') as f :
    for line in f.readlines() :
        data.append(line.strip('\n'))

with open("test.txt", 'r') as f :
    for line in f.readlines() :
        test_data.append(line.strip('\n'))

# GUT CHECK
print(test_data, len(test_data))
print(data[0], data[1], data[-1], len(data))

def part_one(data) :
    horizontal = 0
    depth = 0
    for i in data :
        direction, num = i.split()
        if direction == 'forward' :
            horizontal += int(num)
        elif direction == 'up' :
            depth -= int(num)
        else :
            depth += int(num)
        
    return horizontal * depth

def part_two(data) :
    horizontal = 0
    depth = 0
    aim  = 0
    for i in data :
        direction, num = i.split()
        if direction == 'forward' :
            horizontal += int(num)
            depth += aim * int(num)
        elif direction == 'up' :
            aim -= int(num)
        else :
            aim += int(num)
        
    return horizontal * depth

if __name__ == '__main__' :
    print("PART 1:", part_one(data))
    print("PART 1 TEST INPUT:", part_one(test_data))
    print("PART 2:", part_two(data))
    print("PART 2 TEST:", part_two(test_data))